indy_node/server/node.py

```diff
@@ -296,22 +308,54 @@ def processRequest(self, request: Request, frm: str):
             self.process_query(request, frm)
             self.total_read_request_number += 1
         else:
+            if request.operation[TXN_TYPE] == POOL_RESTART:
+                reply = {}
+                try:
+                    self.configReqHandler.validate(request)
+                    reply = self.generate_action_result(request)
+                    if not self.isProcessingReq(*request.key):
+                        self.startedProcessingReq(*request.key, frm)
+                except Exception as ex:
+                    reply = self.generate_action_result(request,
+                                                        False,
+                                                        ex.args[0])
+                    logger.warning("Restart is failed")
+                finally:
+                    self.sendReplyToClient(reply,
+                                           (request.identifier, request.reqId))
```

Comment:
Creating and sending a reply to the client must be performed after the restart is scheduled (for `action=start`) or cancelled (for `action=cancel`).

In case `action=start` the node need to have some short operability time before the restart in order to send the reply to the client. For this, check in the request validation that if `DATETIME` value is provided then it is greater than or equal to the current time + some time delta. Let's this time delta will be 5 seconds. If `DATETIME` value is not provided then schedule the restart to the current time + this time delta.

Reply:
done